server/src/main/scala/services/crunch/CrunchSystem.scala

```diff
@@ -62,6 +62,11 @@ object CrunchSystem {
 
   val log: Logger = LoggerFactory.getLogger(getClass)
 
+  def crunchStartWithOffset(offsetMinutes: Int)(minuteInQuestion: SDateLike): SDateLike = {
```

Comment:
Offset minutes doesn't imply earlier or later to me, it's a positive number that you are subtracting from midnight... Perhaps minutesBeforeMidnight would be a better name?